### PR TITLE
review & update "Common Toolchains" page + add graph for foss/intel common toolchains and their subtoolchains

### LINKED ISSUE #150: review common-toolchains page

### PR DESCRIPTION
**Pull request by deniskristak**
Revised "Common Toolchains" page + added a graph that explains Toolchains interdependence.
Also added a dependency "Mermaid" as it creates nice diagrams.

edit: fixes #150

### CODE REVIEW

**Comment by deniskristak** [comment]
@boegel  where should I fit info about `CUDA`?

File: docs/common-toolchains.md [hunk 1]
```diff
@@ -31,10 +31,29 @@ The intention is to revise/update the definitions of the common
...
+
+graph LR
+  A[GCCCore] --> |binutils| B[GCC];
+  A --> |binutils| C[iccifort]
+  B --> |MPI libs| E[gompi];
+  C --> |MPI libs| F[iimpi];
+  B --> |Blas, FFTW| D[gfbf];
+  E --> |Blas, FFTW, Lapack libs| G[foss];
+  F --> |Blas, FFTW, Lapack libs| H[intel];
+```
```
Comment (boegel) [inline docs/common-toolchains.md:51]: improved graph:

``` mermaid
graph LR
  A[GCCCore] --> |binutils| B[GCC];
  A --> |binutils| C[iccifort]
  B --> |OpenMPI| E[gompi];
  C --> |impi| F[iimpi];
  B --> |FlexiBLAS + FFTW| D[gfbf];
  D --> |OpenMPI| G[foss];
  E --> |FlexiBLAS + FFTW| G[foss];
  F --> |imkl| H[intel];
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: docs/common-toolchains.md (hunk 1 above)
Comment (boegel) [reply docs/common-toolchains.md:51]: * should have different graph for old/new toolchains
* iccifort should be intel-compilers in recent toolchains
* iimkl subtoolchain is missing in intel/2022b and newer

**Comment by deniskristak** [comment]
https://github.com/easybuilders/easybuild-docs/issues/150

File: docs/common-toolchains.md [hunk 2]
```diff
@@ -31,10 +31,46 @@ The intention is to revise/update the definitions of the common
...
 and `intel`; see below for more details, and also
 [Overview of common toolchains][common_toolchains_overview].
 
+### Toolchain diagram {: #toolchains_diagram }
+
+To be more helpful in understanding the differences between these families, here is a diagram that explains what is added in
+each additional layer.
+
+Note: because there have been a few changes in toolchains, there are notes below the diagram
+that explain the differences between the generations going back to `2021a`.
```
Comment (boegel) [inline docs/common-toolchains.md:44]:
```suggestion
that explain the differences between the generations going back to the `2020b` version of the `foss` and `intel` toolchains.

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]

**Pull request merged**